### PR TITLE
Unmarshal nicks in texts of PRIVMSG and NOTICE from downstreams

### PR DESCRIPTION
**Pull request by delthas**
When writing a PRIVMSG or NOTICE on a channel, it is very common to use
autocompletion to mention other users on that channel. When using soju
in multi-network mode, all users will have their nicked suffixed by
`/network`. This suffix should be removed before sending it upstream.

This adds support for removing all `/network` suffixes in messages sent
to a channel of that network.

### CODE REVIEW

**Comment by delthas** [comment]
Very useful to me. Adding some smarter parsing later would be nice, to avoid URLs being modified if they happen to contain `/network` in their path.

File: downstream.go [hunk 1]
```diff
@@ -546,6 +546,14 @@ func unmarshalUsername(rawUsername string) (username, client, network string) {
 	return username, client, network
 }
 
+func (dc *downstreamConn) unmarshalNicks(uc *upstreamConn, text string) string {
```
Comment (emersion) [inline downstream.go:549]: Can we rename this to e.g. `unmarshalText`? `unmarshalNicks` doesn't make it clear this accepts an arbitrary string.

Comment (emersion) [inline downstream.go:549]: Can we move this right under `unmarshalEntity`, since it's a related operation?

**Comment by delthas** [comment]
Updated.

**Pull request merged**